example/selection.js

```diff
@@ -743,8 +758,118 @@ function updateSelection() {
 
 }
 
+/**
+ * Produce a list of 3D points representing vertices of the box.
+ *
+ * @param {THREE.Box3} box
+ * @param {Array<THREE.Vector3>} target Array of 8 vectors to write to
+ * @returns {Array<THREE.Vector3>}
+ */
+function extractBoxVertices( box, target ) {
+
+	const { min, max } = box;
+	let index = 0;
+
+	for ( let x = 0; x <= 1; x ++ ) {
+
+		for ( let y = 0; y <= 1; y ++ ) {
+
+			for ( let z = 0; z <= 1; z ++ ) {
+
+				const v = target[ index ];
+				v.x = x === 0 ? min.x : max.x;
+				v.y = y === 0 ? min.y : max.y;
+				v.z = z === 0 ? min.z : max.z;
+				index ++;
+
+			}
+
+		}
+
+	}
+
+	return target;
+
+}
+
+/**
+ * Determine if a line segment is to the right of a box.
+ *
+ * @param {THREE.Line3} segment
+ * @param {number} minX The leftmost X coordinate of the box
+ * @param {number} minY The bottommost Y coordinate of the box
+ * @param {number} maxY The topmost Y coordinate of the box
+ * @returns {boolean}
+ */
+function isSegmentToTheRight( segment, minX, minY, maxY ) {
+
+	const sx = segment.start.x;
+	const sy = segment.start.y;
+	const ex = segment.end.x;
+	const ey = segment.end.y;
+
+	if ( sx < minX && ex < minX ) return false;
+	if ( sy > maxY && ey > maxY ) return false;
+	if ( sy < minY && ey < minY ) return false;
+
+	return true;
+
+}
```

Comment:
I think these standalone functions can be moved to a utilities file - or the same location as the "updateSelection" function if they're used there.

Reply:
Yep, I moved this to a separate file along with the function that computes the selection